### PR TITLE
Distance walker client

### PR DESCRIPTION
**Pull request by Aychihuahuausa**
Add storage to save distance that walker spend

### CODE REVIEW

File: src/app/main-client/main-client.component.ts [hunk 1]
```diff
@@ -169,9 +173,11 @@ export class MainClientComponent implements OnInit {
       }
     })
     this.mapService.mapLoaded.subscribe(() => {
-      this.userService.getCurrentWalker().then(x => {
-        this.getCurrentPostion()
-      });
+      if (!this.userService.currentWalker) {
```
Comment (heshambadawy) [inline src/app/main-client/main-client.component.ts:176]: Supposed not to reached here , if he isn't  already authorized

File: src/app/main-client/main-client.component.ts [hunk 2]
```diff
@@ -199,7 +207,7 @@ export class MainClientComponent implements OnInit {
         if (x) {
           this.mainClientService.getClientDogsImagesUrl();
           this.clientDog = { lat: x.clientLat, lng: x.clientLong };
-          if (x.statusId !== RequestStatus.Cancel) {
+          if (x.statusId !== RequestStatus.Cancel && x.statusId !== RequestStatus.Finished) {
```
Comment (heshambadawy) [inline src/app/main-client/main-client.component.ts:210]: Don't use the A b c names

File: src/app/main-client/walk-started/walk-started.component.ts [hunk 3]
```diff
@@ -35,7 +36,7 @@ export class WalkStartedComponent implements OnInit {
   syncInterval: any;
   constructor(private cordovaService: CordovaService, public mainClientService: MainClientService,
```
Comment (heshambadawy) [inline src/app/main-client/walk-started/walk-started.component.ts:37]: Add space before any method "cc"

File: src/app/share/map/map.service.ts [hunk 4]
```diff
@@ -178,9 +178,10 @@ export class MapService {
         travelMode: google.maps.TravelMode.DRIVING,
       });
   }
-  distanceInKmBetweenEarthCoordinates = function (location1: Position, location2: Position) {
+  distanceInKmBetweenEarthCoordinates = function (loc1: Position, loc2: Position) {
     var earthRadiusKm = 6371;
-
+    let location1 = { ...loc1 };
```
Comment (heshambadawy) [inline src/app/share/map/map.service.ts:183]: it's already object why u need spread it

File: src/app/share/map/map.service.ts [hunk 5]
```diff
@@ -178,9 +178,10 @@ export class MapService {
         travelMode: google.maps.TravelMode.DRIVING,
       });
   }
-  distanceInKmBetweenEarthCoordinates = function (location1: Position, location2: Position) {
```
Comment (heshambadawy) [inline src/app/share/map/map.service.ts:181]: Is your testing assure that getting distance come in approximately accurate matter?
Need unit testing for that to let me handle it later and all similar "Todo Add "

**Review by heshambadawy — CHANGES_REQUESTED** [review]

File: src/app/main-client/main-client.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (heshambadawy) [reply src/app/main-client/main-client.component.ts:176]: ToDo later

Comment (heshambadawy) [reply src/app/main-client/main-client.component.ts:176]: Done, later consideration

File: src/app/main-client/main-client.component.ts (hunk 2 above)
Comment (heshambadawy) [reply src/app/main-client/main-client.component.ts:210]: renaming

File: src/app/share/map/map.service.ts (hunk 4 above)
Comment (heshambadawy) [reply src/app/share/map/map.service.ts:183]: It's Done

File: src/app/main-client/main-client.component.ts [hunk 6]
```diff
@@ -169,9 +173,11 @@ export class MainClientComponent implements OnInit {
...
-        this.getCurrentPostion()
-      });
+      if (!this.userService.currentWalker) {
+        this.userService.getCurrentWalker().then(x => {
+          this.getCurrentPostion()
+        });
+      }
     })
   }
   async getCurrentPostion() {
```
Comment (heshambadawy) [inline src/app/main-client/main-client.component.ts:183]: add watch position after getCurrentPoistion , and remove redenecy of code lines

**Review by heshambadawy — CHANGES_REQUESTED** [review]

File: src/app/main-client/main-client.component.ts [hunk 7]
```diff
@@ -124,6 +127,7 @@ export class MainClientComponent implements OnInit {
               modal.dismiss();
               this.mapService.clearMap();
```
[the remark this answers is not in the document]
Comment (Aychihuahuausa) [reply src/app/main-client/main-client.component.ts:128]: done

File: src/app/main-client/main-client.component.ts (hunk 6 above)
Comment (Aychihuahuausa) [reply src/app/main-client/main-client.component.ts:173]: done

File: src/app/main-client/main-client.component.ts (hunk 2 above)
Comment (Aychihuahuausa) [reply src/app/main-client/main-client.component.ts]: done

File: src/app/main-client/walk-started/walk-started.component.ts (hunk 3 above)
Comment (Aychihuahuausa) [reply src/app/main-client/walk-started/walk-started.component.ts:39]: done

**Review by heshambadawy — APPROVED** [review]

**Pull request merged**